src/display/lv_display.c

```diff
@@ -396,6 +396,26 @@ void lv_display_set_draw_buffers(lv_display_t * disp, lv_draw_buf_t * buf1, lv_d
     disp->buf_act = disp->buf_1;
 }
 
+void lv_display_set_buffers(lv_display_t * disp, void * buf1, void * buf2, uint32_t buf_size,
+                            lv_display_render_mode_t render_mode)
+{
+    uint32_t w = lv_display_get_horizontal_resolution(disp);
+    uint32_t h = lv_display_get_vertical_resolution(disp);
+    LV_ASSERT(w != 0 && h != 0);
+
+    lv_color_format_t cf = lv_display_get_color_format(disp);
+    uint32_t stride = lv_draw_buf_width_to_stride(w, cf);
+    if(stride * h > buf_size && render_mode != LV_DISPLAY_RENDER_MODE_PARTIAL) {
+        LV_ASSERT_MSG(false, "buffer too small for non partial render mode.");
```

Comment:
```suggestion
        LV_ASSERT_MSG(false, "the buffer needs to be large enough for the full screen in %s mode.", render_mode == LV_DISPLAY_RENDER_MODE_FULL ? "FULL" : "DIRECT");
```

Reply:
How about this one in order to keep log as short as possible, considering flash size?
```c
        LV_ASSERT_MSG(false, "buffer too small for full screen in %s mode.",
                      render_mode == LV_DISPLAY_RENDER_MODE_FULL ? "FULL" : "DIRECT");
```

Reply:
This one?
```c
LV_ASSERT_MSG(false, "%s mode requires screen sized buffer(s)",
                      render_mode == LV_DISPLAY_RENDER_MODE_FULL ? "FULL" : "DIRECT");
```